src/controllers/delegates/midioptionsdelegate.cpp

```diff
@@ -66,28 +106,72 @@ QString MidiOptionsDelegate::displayText(const QVariant& value,
-
-    QComboBox* pComboBox = qobject_cast<QComboBox*>(editor);
-    if (pComboBox == nullptr) {
+    auto* pComboBox = qobject_cast<QComboBox*>(editor);
+    if (!pComboBox) {
         return;
     }
-    for (int i = 0; i < pComboBox->count(); ++i) {
-        if (MidiOptions(pComboBox->itemData(i).toInt()) & options) {
-            pComboBox->setCurrentIndex(i);
-            return;
+
+    // Update checked states
+    const MidiOptions options = index.data(Qt::EditRole).value<MidiOptions>();
+    const auto* pModel = static_cast<QStandardItemModel*>(pComboBox->model());
+    DEBUG_ASSERT(pModel);
+    for (int row = 0; row < pModel->rowCount(); row++) {
+        auto pItem = pModel->item(row, 0);
+        if (!pItem->isCheckable()) {
+            continue;
         }
+        auto opt = static_cast<MidiOption>(pItem->data().toInt());
+        pItem->setCheckState(options.testFlag(opt) ? Qt::Checked : Qt::Unchecked);
     }
+
+    // Show popup immediately, as with the other editors, no 'edit' click
+    // required to open the list view.
+    pComboBox->showPopup();
 }
 
 void MidiOptionsDelegate::setModelData(QWidget* editor,
                                        QAbstractItemModel* model,
                                        const QModelIndex& index) const {
-    MidiOptions options;
-    QComboBox* pComboBox = qobject_cast<QComboBox*>(editor);
-    if (pComboBox == nullptr) {
+    // Collect checked options and write them back to the model
+    auto* pComboBox = qobject_cast<QComboBox*>(editor);
+    if (!pComboBox) {
         return;
     }
-    options = MidiOptions(pComboBox->itemData(pComboBox->currentIndex()).toInt());
+
+    const auto* pModel = static_cast<QStandardItemModel*>(pComboBox->model());
+    DEBUG_ASSERT(pModel);
+    MidiOptions options;
+    for (int i = 0; i < pModel->rowCount(); i++) {
+        auto* pItem = pModel->item(i);
+        // Only check for Qt::Checked or else, ignore Qt::PartiallyChecked
+        options.setFlag(static_cast<MidiOption>(pItem->data().toUInt()),
+                pItem->checkState() == Qt::Checked);
+    }
     model->setData(index, QVariant::fromValue(options), Qt::EditRole);
 }
+
+void MidiOptionsDelegate::commitAndCloseEditor(int index) {
+    QComboBox* pComboBox = qobject_cast<QComboBox*>(sender());
+    if (!pComboBox) {
+        return;
+    }
+    const auto* pModel = static_cast<QStandardItemModel*>(pComboBox->model());
+    DEBUG_ASSERT(pModel);
+    auto* pItem = pModel->item(index);
+    DEBUG_ASSERT(pItem);
+    if (pItem->isCheckable()) {
+        pItem->setCheckState(pItem->checkState() == Qt::Checked ? Qt::Unchecked : Qt::Checked);
+    } else {
+        // Clear was selected. Uncheck all other items
+        for (int row = 0; row < pModel->rowCount() - 1; row++) {
+            if (row == index) { // Actually it's the last item, but this is safer.
+                continue;
+            }
+            auto pItem = pModel->item(row, 0);
```

Comment:
```suggestion
            auto *pItem = pModel->item(row, 0);
```

Reply:
actually just `pItem = pModel->item(row, 0);` because it's already declared a few lines above.